apis/externalsecrets/v1/generic_store.go

```diff
@@ -41,6 +42,10 @@ type GenericStore interface {
 	GetStatus() SecretStoreStatus
 	SetStatus(status SecretStoreStatus)
 	Copy() GenericStore
+
+	AddFinalizer(finalizer string)
+	RemoveFinalizer(finalizer string)
```

Comment:
I think you do not need these methods at all because you can simply work with metav1.Object